### ISSUE TITLE
Array slicing with index grater than 65,535 not working.

### ISSUE THREAD

**Issue by gkutiel** [stedolan/jq#2164]
**Describe the bug**
The array slicing operation not working for indexes larger than 65,535

**To Reproduce**
Run the following commands
```seq 0 65537 | jq '[inputs][65535:]'```
```seq 0 65537 | jq '[inputs][65536:]'```

**Expected behavior**
Output for the first command is correct: ```[65536, 65537]```
Output for the second command is incorrect: ```[1]``` where it should be ```[65537]```

**Environment (please complete the following information):**
 - Ubuntu
 - jq version jq-1.5-1-a5b5cbe

**Additional context**

**Comment by pkoppstein** [comment]
This was a bug that was fixed in the official release of jq 1.5 so perhaps your jq is a pre-release version.

If at all reasonable, I'd suggest you upgrade to jq 1.6 or, with very minimal risk and quite possibly very rewarding benefits, the master version.

At any rate, on a Mac:
```
$ seq 0 65537 | jq-1.4 -s -c '.[65536:]'
[0,1]

$ seq 0 65537 | jq-1.5 -sc '.[65536:]'
[65536,65537]

$ seq 0 65537 | jq-1.6 -sc '.[65536:]'
[65536,65537]
```

**Comment by gkutiel** [comment]
Works with jq version 1.6.
Thank you.
